### DISCUSSION TITLE
Problem with displaying commits in contributions graph

### DISCUSSION THREAD

**Discussion by MontyHubert** [community/community#80792] [New to GitHub] [answered]
Hello, I have a problem with the display of commits in the contribution graph.
I am using SHH connection with github, on my profile I can only see one commit on my private repo, when Im inside my repo i can see all commits.
It also does not update my commits in the contribution chart.
Do you have any idea or way how I can fix this?
Thank you in advance

**Comment by varunpats** [comment]
Hey, commits will appear on your contributions graph if they meet all of the following conditions:

1. The email address used for the commits is associated with your account on GitHub.com.
2. The commits were made in a standalone repository, not a fork.
3. The commits were made:
- In the repository's default branch
- In the gh-pages branch (for repositories with project sites)

Hope this helps to find out what is the cause. thanks!

**Comment by AnujaDilrukshiHerath** [comment, marked as answer]
Check Repository Privacy Settings:

Ensure that your repository is set to be private if that's the intended setting. If it's public, your commits should be reflected in your contribution graph.
Verify Commit Author Email:
Ensure that the email address associated with your commits matches the email linked to your GitHub account. You can check this in your commit history.
bash
Copy code
git log

If the email addresses don't match, you may need to configure your local Git settings to use the correct email:
bash
Copy code
git config --global user.email "your-email@example.com"
Check SSH Key Configuration:

Confirm that your SSH key is correctly configured and associated with your GitHub account. You can add an SSH key in your GitHub account settings.
Update Repository Remotes:
Confirm that your local repository has the correct remote URL. You can check and update it using:

bash
Copy code
git remote -v
git remote set-url origin git@github.com:username/repository.git
Force Push:

If you've made changes to your commit history locally, you might need to force push to update the remote repository:
bash
Copy code
git push -f origin branch-name
Be cautious with force pushing, especially if collaborating with others, as it rewrites history.

Wait for GitHub to Update:
Sometimes, GitHub's contribution graph takes some time to update. Give it some time, and check later to see if the commits appear.

**Reply by AryanshuCreates** [reply]
how long does it take?

**Comment by MontyHubert** [comment]
Thank you all for your help, I made a mistake in the global configuration.
Thanks again, I wish you all the best :) !
